src/resolve/resolved-varlink.c

```diff
@@ -530,6 +553,516 @@ static int vl_method_resolve_address(Varlink *link, JsonVariant *parameters, Var
         return 1;
 }
 
+static int append_txt(JsonVariant **ret_txt, DnsResourceRecord *rr) {
+        int r;
+
+        assert(rr);
+        assert(rr->key);
+
+        if (rr->key->type != DNS_TYPE_TXT)
+                return 0;
```

Comment:
you return from the function here successfully (i.e. with an error code >= 0).

according to our coding style functions should initialize all return paramaters on success (and leave them untouched on failure). THus, you need to add `*ret_txt = NULL` here, if you return 0.